dashboard/src/components/AppView/AppView.tsx

```diff
@@ -266,56 +291,77 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
     }
   }
 
-  private deploymentArray(): IResource[] {
-    return Object.keys(this.state.deployments).map(k => this.state.deployments[k]);
+  // Retrieve the deployments/service/ingresses if they are already loaded
+  private arrayFromState(type: string): IResource[] {
+    const elems = Object.keys(this.state[type]);
+    const res: IResource[] = [];
+    elems.forEach(e => {
+      if (this.state[type][e].resource) {
+        res.push(this.state[type][e].resource);
+      }
+    });
+    return res;
+  }
+
+  // Retrieve is some deployments/service/ingresses is still loading
+  private isLoading(type: string) {
+    return _.some(this.state[type], e => e.isFetching);
   }
 
   private deleteApp = (purge: boolean) => {
     return this.props.deleteApp(this.props.releaseName, this.props.namespace, purge);
   };
 
-  private filterByKind = (kind: string) => {
-    const kubeItems = _.pickBy(this.props.resources, r => {
-      return r.item && r.item.kind === kind;
-    });
-    const res = {};
-    _.each(kubeItems, (i, k) => {
-      res[k] = i.item;
+  private filterResourceByType = (type: string) => {
+    return _.pickBy(this.props.resources, (r, k) => {
+      return k.indexOf(`/${type}/`) > -1;
     });
-    return res;
   };
 
-  private findErrorByResourceType = (type: string): IError | null => {
-    const kubeItems = _.pickBy(this.props.resources, (r, k) => {
-      return k.indexOf(`/${type}/`) > -1;
-    });
+  private findError = (resources: { [s: string]: IKubeItem }): IError | null => {
     let error = null;
-    _.each(kubeItems, (i, k) => {
+    _.each(resources, (i, k) => {
       if (i.error) {
         error = { resource: k, error: i.error };
       }
     });
     return error;
   };
 
+  private findLoadingResource = (resources: { [s: string]: IKubeItem }): boolean => {
```

Comment:
Could you explain why is this function needed?

Reply:
it's convenient to find a loading item from a list. I've moved this to a helper function.